vamb/reclustering.py

```diff
@@ -0,0 +1,319 @@
+"""
+The following code is a modification of a k-means based reclustering algorithm first published at https://github.com/BigDataBiology/SemiBin
+The original code is distributed under MIT License.
+"""
+
+import os
+import subprocess
+import tempfile
+import sys
+import contextlib
+from typing import IO
+from sklearn.cluster import KMeans
+import numpy as np
+from collections import defaultdict
+import pandas as pd
+
+
+def log(string: str, logfile: IO[str], indent: int = 0):
+    print(("\t" * indent) + string, file=logfile)
+    logfile.flush()
+
+
+def fasta_iter(fname, full_header=False):
```

Comment:
You might want to use `vamb.vambtools.byte_iterfasta` and `vamb.vambtools.Reader` here instead, for simplicity (and it's a pretty good implementation)